languages/kotlin/sdk/build.gradle

```diff
@@ -42,14 +41,25 @@ publishing {
     publications {
         maven(MavenPublication) {
             groupId = 'com.bitwarden'
-            artifactId = 'sdk'
+            artifactId = 'sdk-android'
 
-            def details = versionDetails()
+            // Determine the version from the git history.
+            //
+            // PRs: use the branch name.
+            // Master: Grab it from `crates/bitwarden/Cargo.toml`
```

Comment:
🎨 I'd like to discuss the approach on my PR to use non-snapshots for `master` builds and to also include the version before the branch name otherwise. These seem like best practices with Maven versioning. Do you want to use another workflow for non-snapshot builds?

Reply:
Oh also, when we entertain more automated versioning (perhaps not in this repo, but it's happening elsewhere) this could very much be driven inside the Actions workflows so that's why I copied over what the Rust builds were doing. Using that and passing in the version as a step dependency seems like the long-term solution to me vs. individual implementations figuring it out.

Reply:
Everything except tagged versions should be snapshots right? A non snapshot version should be static and never modified.

Reply:
Per recommendations at https://stackoverflow.com/a/11414236/553125 it seems like a good idea to have both version and branch name.

Reply:
After reading the comments to that answer and the other answers.

Can you verify that both maven and gradle won't prefer 0.3.0-sdk-android-pas-maven over 0.3.0? The specification is a bit confusing https://maven.apache.org/pom.html#version-order-specification.

```
$ java -jar maven-artifact-3.9.4.jar 0.3.0-sdk-android-ps-maven 0.3.0

Display parameters as parsed by Maven (in canonical form and as a list of tokens) and comparison result:
1. 0.3.0-sdk-android-ps-maven -> 0.3-sdk-android-ps-maven; tokens: [0, 3, [sdk, [android, [ps, [maven]]]]]
   0.3.0-sdk-android-ps-maven > 0.3.0
2. 0.3.0 -> 0.3; tokens: [0, 3]
```